### ISSUE TITLE
Suggested cop: Warn of unnecessary `initialize` methods

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#10438]
Sometimes a refactor can lead to a situation such as:

```ruby
class Foo < Bar
  def initialize(a, b)
    super
  end
end
```

**Comment by vlad-pisanov** [comment]
This might be unsafe / difficult to implement if default values are used in the arguments:

```ruby
class Parent
  def initialize(a, b)
    puts [a, b]
  end
end

class Child < Parent
  def initialize(a, b = 42)  # the signature has changed; can't safely remove this definition
    super
  end
end
```

**Comment by andyw8** [comment]
Good point, the cop should ignore if there are any defaults (positional or keyword, i.e. `initialize(a, b: 42)`).

**Comment by dvandersluis** [comment]
@andyw8 @vlad-pisanov I opened a PR to implement this, please take a look.
